operator/pkg/lbipam/lbipam.go

```diff
@@ -0,0 +1,1807 @@
+				}
+
+				// Requeue if the error still isn't resolved.
+				poolRetryQueue.Add(retry)
+			}
+		}
+	}
+
+	return nil
+}
+
+func (ipam *LBIPAM) initSyncSvc(svcChan <-chan svcEvent) error {
+	svcSync := false
+	for !svcSync {
+		select {
+		case event := <-svcChan:
+			event.Handle(
+				func() error {
+					svcSync = true
+
+					err := ipam.satisfyServices()
+					if err != nil {
+						// Let the resource retry after the initialization
+						return err
+					}
+
+					err = ipam.updateAllPoolCounts()
+					if err != nil {
+						// Let the resource retry after the initialization
+						return err
+					}
+
+					return nil
+				},
+				func(k resource.Key, svc *slim_core_v1.Service) error {
+					err := ipam.handleUpsertService(svc)
+					if err != nil {
+						// Let the resource retry after the initialization
+						return err
+					}
+
+					return nil
+				},
+				func(k resource.Key, svc *slim_core_v1.Service) error {
+					ipam.logger.Error("Got a Service deleted event while syncing")
+					return nil
+				},
+			)
+		}
+	}
+
+	return nil
+}
+
+func (ipam *LBIPAM) poolOnUpsert(k resource.Key, pool *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) error {
+	// Deep copy so we get a version we are allowed to update the status
+	pool = pool.DeepCopy()
+
+	var err error
+	if _, exists := ipam.pools[pool.GetName()]; exists {
+		ipam.logger.Debugf("Updating pool '%s'", pool.GetName())
+		err = ipam.handlePoolModified(pool)
+		if err != nil {
+			ipam.logger.WithError(err).Error("Error in handlePoolModified()")
+		}
+	} else {
+		ipam.logger.Debugf("Adding pool '%s'", pool.GetName())
+		err = ipam.handleNewPool(pool)
+		if err != nil {
+			ipam.logger.WithError(err).Error("Error in handleNewPool()")
+		}
+	}
+	if err != nil {
+		return err
+	}
+
+	err = ipam.settleConflicts()
+	if err != nil {
+		ipam.logger.WithError(err).Error("Error in settleConflicts()")
+		return fmt.Errorf("settleConflicts: %w", err)
+	}
+
+	err = ipam.satisfyAndUpdateCounts()
+	if err != nil {
+		ipam.logger.WithError(err).Error("Error in satisfyAndUpdateCounts()")
+		return fmt.Errorf("satisfyAndUpdateCounts: %w", err)
+	}
+
+	return nil
+}
+
+func (ipam *LBIPAM) poolOnDelete(k resource.Key, pool *cilium_api_v2alpha1.CiliumLoadBalancerIPPool) error {
+	err := ipam.handlePoolDeleted(pool)
+	if err != nil {
+		ipam.logger.WithError(err).Error("Error in handlePoolDeleted()")
+		return fmt.Errorf("handlePoolDeleted: %w", err)
+	}
+
+	err = ipam.settleConflicts()
+	if err != nil {
+		ipam.logger.WithError(err).Error("Error in settleConflicts()")
+		return fmt.Errorf("settleConflicts: %w", err)
+	}
+
+	err = ipam.satisfyAndUpdateCounts()
+	if err != nil {
+		ipam.logger.WithError(err).Error("Error in satisfyAndUpdateCounts()")
+		return fmt.Errorf("satisfyAndUpdateCounts: %w", err)
+	}
+
+	return nil
+}
+
+func (ipam *LBIPAM) svcOnUpsert(k resource.Key, svc *slim_core_v1.Service) error {
+	ipam.logger.Debugf("Added/updated service '%s/%s'", svc.GetNamespace(), svc.GetName())
```

Comment:
`Debugf` is quite expensive even if it doesn't end up being printed. Consider removing these from the final version and add metrics instead.